### DISCUSSION TITLE
io.BytesIO as modules to Pyinstaller.__main__.run()

### DISCUSSION THREAD

**Discussion by farnking** [pyinstaller/pyinstaller#5594] [Features & Development]
Hi, dear Sirs: 

I am developing a tool in python and would like to see whether we can use io.BytesIO or io.StringsIO as modules to Pyinstaller.__main__.run().
The motivation is protecting my IP and in the same time maintain the portability of our tool across platforms.  
To make the download cross-platform, our server may send pyc bits to the clients, then convert those pyc files-in-memory to pyd files-in-memory, and then finally use pyinstaller to package everything to EXE with a desktop icon.  
In this way, all the pyc and py files downloaded from our server only stay in memory without a trace in the clients' disks.  
We feel this is the best way to protect our IP with your pyinstaller and python.  
Moreover, we don't have to worry about the generation of native machine code in our server side. 

But all the related pages in the internet only show how to use pyinstaller.__main__.run() for using py (or maybe pyc) files in disks as modules.
In this way, my tool pyc file may be copied in the clients' folder and decompiled for the source python program of my tool.
It is said that decompiling pyc files is much easier than decompiling their pyd counterparts.
Thus I am curious if it is possible to run pyinstaller.main.run() with io.stringsIO or io.bytesIO as the source modules.

Thanks

Farn

**Comment by bwoodsend** [comment]
Well it certainly should be but isn't. Even our own test suite is resorts to temporary files.

**Reply by farnking** [reply]
OK!  Thanks for the reply!  But I would appreciate it very much if you can put this feature into recent schedule.  
I believe this will greatly enhance people's confidence in protecting their IPs with Python and Pyinstaller while enjoying the portability of Python.  
Looking forward to it.
